swayidle/main.c

```diff
@@ -0,0 +1,448 @@
+		exit(-1);
+	}
+	struct swayidle_timeout_cmd *cmd =
+		calloc(1, sizeof(struct swayidle_timeout_cmd));
+	cmd->timeout = seconds*1000;
+
+	wlr_log(L_DEBUG, "Register idle timeout at %d ms", cmd->timeout);
+	wlr_log(L_DEBUG, "Setup idle");
+	cmd->idle_cmd = parse_command(argc - 2, &argv[2]);
+
+	int result = 4;
+	if (argc >= 7 && !strcmp("resume", argv[4])) {
+		wlr_log(L_DEBUG, "Setup resume");
+		cmd->resume_cmd = parse_command(argc - 5, &argv[5]);
+		result = 7;
+	}
+	list_add(state.timeout_cmds, cmd);
+	return result;
+}
+
+int parse_args(int argc, char *argv[]) {
+	int c;
+	char *endptr;
+	static uint32_t idle_timeout = 0, lock_timeout = 0;
+	char *lock_timeout_str = NULL;
+	char *idle_timeout_str = NULL;
+
+	while ((c = getopt(argc, argv, "i:hl:s:dx:")) != -1) {
+		switch(c) {
+		case 'i':
+			errno = 0;
+			idle_timeout = strtoul(optarg, &endptr, 10);
+			if (errno != 0 || *endptr != '\0') {
+				wlr_log(L_ERROR, "Invalid idle parameter '%s', it should be a \
+numeric value representing seconds", optarg);
+				exit(-1);
+			}
+			idle_timeout_str = optarg;
+			break;
+		case 'l':
+			errno = 0;
+			lock_timeout = strtoul(optarg, &endptr, 10);
+			if (errno != 0 || *endptr != '\0') {
+				wlr_log(L_ERROR, "Invalid lock parameter '%s', it should be a \
+numeric value representing seconds", optarg);
+				exit(-1);
+			}
+			lock_timeout_str = optarg;
+			break;
+		case 'd':
+			debug = true;
+			break;
+		case 'x':
+			lock_cmd = strdup(optarg);
+			break;
+		case 'h':
+		case '?':
+			printf("Usage: %s [OPTIONS]\n", argv[0]);
+			printf("  -i seconds optional\t\tidle timeout in seconds\n");
+			printf("  -l seconds optional\t\tlock timeout in seconds\n");
+			printf("  -x lock_command\t\t to raise the lockscreen on systemd \
+sleep event\n");
+			printf("  -d\t\t\tdebug\n");
+			printf("  -h\t\t\t\tthis help menu\n");
+			return 1;
+		default:
+			return 1;
+		}
+	}
+
+	if (debug) {
+		wlr_log_init(L_DEBUG, NULL);
+		wlr_log(L_DEBUG, "Loglevel debug");
+	} else {
+		wlr_log_init(L_INFO, NULL);
+	}
+
+
+	state.timeout_cmds = create_list();
+
+	if (idle_timeout) {
+		wlr_log(L_DEBUG, "Idle timeout %s", idle_timeout_str);
+		char *params[7];
+		params[0] = "timeout";
+		params[1] = idle_timeout_str;
+		params[2] = "exec";
+		params[3] = "swaymsg \"output * dpms off\"";
+		params[4] = "resume";
+		params[5] = "exec";
+		params[6] = "swaymsg \"output * dpms on\"";
+		parse_timeout(7, params);
+	}
+	if (lock_timeout) {
+		wlr_log(L_DEBUG, "Lock timeout %s", lock_timeout_str);
+		char *params[4];
+		params[0] = "timeout";
+		params[1] = lock_timeout_str;
+		params[2] = "exec";
+		params[3] = "swaymsg exec swaylock -c 000000";
+		parse_timeout(4, params);
+	}
+	int i = optind;
+	while (i < argc) {
+		if (!strcmp("timeout", argv[i])) {
+			wlr_log(L_DEBUG, "Got timeout");
+			i += parse_timeout(argc - i, &argv[i]);
+		} else {
+			wlr_log(L_ERROR, "Unsupported command '%s'", argv[i]);
+			exit(-1);
+		}
+	}
+	return 0;
```

Comment:
If the user doesn't configure anything you should probably warn them with some logging

Reply:
That is done at: 
    if (!should_run) {
        wlr_log(L_INFO, "No command specified! Nothing to do, will exit");
        sway_terminate(0);
    }